budget/app.py

```diff
@@ -1,14 +1,176 @@
 import json
 import logging
 
+import boto3
+from botocore.exceptions import ClientError
+import requests
+
 log = logging.getLogger(__name__)
 log.setLevel(logging.DEBUG)
 
+# TODO configure this URL?
+SYNAPSE_TEAM_MEMBER_URL = 'https://repo-prod.prod.sagebase.org/repo/v1/teamMembers'
```

Comment:
you shouldn't need to reference this URL explicitly.  See the Python command, https://python-docs.synapse.org/build/html/Client.html#synapseclient.Synapse.getTeamMembers

Could the problem be that the Python client prevents you from using it anonymously (without logging in)?

Reply:
That's right, I just call the endpoint directly rather than using the synapse client because authentication puts an additional burden on me here -- creating a "service account" for Synapse is a pain.

Reply:
There should be no need to use a service account to call a service that can be called anonymously.   We need to do https://sagebionetworks.jira.com/browse/SYNPY-1084.  Pending that change I think you are right to 'roll your own' Python client using a Python HTTP client.